go/client/kvstore_api_handler.go

```diff
@@ -86,35 +93,50 @@ func (a *getEntryOptions) Check() error {
 	return nil
 }
 
+func (t *kvStoreAPIHandler) getUsername() (username string, err error) {
+	status, err := t.config.GetCurrentStatus(context.Background(), 0)
+	if err != nil {
+		return username, err
+	}
+	return status.User.Username, nil
+}
+
 func (t *kvStoreAPIHandler) getEntry(ctx context.Context, c Call, w io.Writer) error {
 	var opts getEntryOptions
 	if err := unmarshalOptions(c, &opts); err != nil {
 		return t.encodeErr(c, err, w)
 	}
+	if opts.Team == nil {
+		team, err := t.getUsername()
+		if err != nil {
+			return t.encodeErr(c, err, w)
+		}
+		opts.Team = &team
+	}
 	arg := keybase1.GetKVEntryArg{
 		SessionID: 0,
-		TeamName:  opts.Team,
+		TeamName:  *opts.Team,
 		Namespace: opts.Namespace,
 		EntryKey:  opts.EntryKey,
 	}
-	res, err := t.cli.GetKVEntry(ctx, arg)
+	res, err := t.kvstore.GetKVEntry(ctx, arg)
 	if err != nil {
 		return t.encodeErr(c, err, w)
 	}
 	return t.encodeResult(c, res, w)
 }
 
 type putEntryOptions struct {
-	Team       string `json:"team"`
-	Namespace  string `json:"namespace"`
-	EntryKey   string `json:"entryKey"`
-	Revision   *int   `json:"revision"`
-	EntryValue string `json:"entryValue"`
+	Team       *string `json:"team,omitempty"`
+	Namespace  string  `json:"namespace"`
+	EntryKey   string  `json:"entryKey"`
+	Revision   *int    `json:"revision"`
+	EntryValue string  `json:"entryValue"`
 }
 
 func (a *putEntryOptions) Check() error {
-	if len(a.Team) == 0 {
-		return errors.New("`team` field required")
+	if a.Team != nil && len(*a.Team) == 0 {
+		return errors.New("if setting optional `team` field, it must not be empty")
```

Comment:
not a big deal, but if you don't do this check here, it'll probably just error when it tries to look up the team later in the flow, right?

Reply:
just confirmed that service errors if teamname is "", so i deleted this check